src/hotspot/share/gc/shared/taskqueue.inline.hpp

```diff
@@ -205,7 +205,7 @@ bool OverflowTaskQueue<E, F, N>::pop_overflow(E& t)
 // reads elems[oldAge.top].  The owner's bottom == the thief's oldAge.top.
 // (4) Thief will discard the read value, because its cmpxchg of age will fail.
 template<class E, MEMFLAGS F, unsigned int N>
-bool GenericTaskQueue<E, F, N>::pop_global(E& t) {
+typename GenericTaskQueue<E, F, N>::PopResult GenericTaskQueue<E, F, N>::pop_global(E& t) {
```

Comment:
These are places that are just made for trailing return types, putting the return type in the scope of the function.  That's still in the "undecided features" list in the style guide though.

```
template<class E, MEMFLAGS F, unsigned int N>
auto GenericTaskQueue<E, F, N>::pop_global(E& t) -> PopResult { ... }
```